### DISCUSSION TITLE
How to update previously fetched data (cache) after a PUT request.

### DISCUSSION THREAD

**Discussion by lezoudali** [TanStack/query#256] [General] [answered]
I have a form component to update an organization data, I first fetch the org data
using react-query, like so

```
const { status, data, error } = useQuery(
    ["organization", http://user.organization.id, authToken],
    (key, ...params) => fetchOrganization(...params)
  );
```

the data returned is used to fill out the organization form as initial values to be updated.
I know react-query caches the org data fetched; but when the user submits the form, I make a PUT request to the server which returns the updated org JSON data. Is there a way to update the cached 
organization data which was initially fetched using react-query in the code snippet above?

**Comment by cherniavskii** [comment, marked as answer]
Yes, you can use `queryCache.setQueryData` - see https://github.com/tannerlinsley/react-query#query-updates-from-mutations
